### DISCUSSION TITLE
The memory [M] buttons do not work ...

### DISCUSSION THREAD

**Discussion by Vectord2000** [arch10/Calculator-Plus#51] [Q&A] [answered]
**Describe the bug**

**To Reproduce**
1. tap any number : 55 
2. Swipe left to open the scientific functions
3. Tap on [MS] and the value "55" is not stored : 
4. swipe right to hide the menu and clear the screen from any values. 
5. Swipe left again to show scientific functions
6. Tap on [MR] and the value "55" will not show. 
7. Tapping [M+] and/or [M-] does not do anything. 

**Expected behavior**
The memory [M] buttons [MS], [MR], [M+] and [M-] buttons should store, restore, add and deduct the actual value displayed on screen, which is not happening. 

**Android (please complete the following information):**
 - Device: [Samsung Galaxy S20+ 5G]
 - OS: Android 13, One UI 5.1 
 - App Version : 2.4.4

**Comment by arch10** [comment, marked as answer]
@Vectord2000 Memory functions would work when there is expression in the result state. As per your steps, since you havn't started any calculation with number 55, there is no value in storing it into [MS]. Try doing this:

1. write: 55 + 2
2. Open advanced panel, tap [MS] and then close the panel. The result of the expression, i.e., 57 is copied over to the memory.
3. Clear all the expressions.
4. Open advanced panel, tap [MR]. This will restore the number 57 from the memory.
5. You can similarly try using [M+] & [M-]
